lib/schema.js

```diff
@@ -615,6 +615,12 @@ Schema.prototype.add = function add(obj, prefix) {
     if (key === '_id' && val === false) {
       continue;
     }
+    // Deprecate setting schema paths to primitive types (gh-7558)
+    if (key !== '_id' && typeof val === 'boolean') {
```

Comment:
👍 I've added an extra case to handle primitives specified as strings, otherwise, tests like [this](https://github.com/lpizzinidev/mongoose/blob/8123ef39830bf3f5ee27f07029765d00ebc1bdd5/test/schema.test.js#L171-L201) were failing.